addons/rating/models/rating_mixin.py

```diff
@@ -42,12 +47,38 @@ def _compute_rating_percentage_satisfaction(self):
                 grades_per_parent[parent_id]['okay'] += item['__count']
             else:
                 grades_per_parent[parent_id]['bad'] += item['__count']
+            rating_scores_per_parent[parent_id] += rating * item['__count']
 
         # compute percentage per parent
         for record in self:
             repartition = grades_per_parent.get(record.id, default_grades)
-            record.rating_count = sum(repartition.values())
-            record.rating_percentage_satisfaction = repartition['great'] * 100 / sum(repartition.values()) if sum(repartition.values()) else -1
+            rating_count = sum(repartition.values())
+            record.rating_count = rating_count
+            record.rating_percentage_satisfaction = repartition['great'] * 100 / rating_count if rating_count else -1
+            record.rating_avg = rating_scores_per_parent[record.id] / rating_count if rating_count else 0
+            record.rating_avg_percentage = record.rating_avg / 5
+
+    def _search_rating_avg(self, operator, value):
+        if operator not in ['=', '!=', '<=', '<', '>', '>=']:
+            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
+        if not isinstance(value, (int, float)):
+            raise NotImplementedError('The value must be an integer or a float value.')
+        where_write_date_query = ''
+        if self._rating_satisfaction_days and isinstance(self._rating_satisfaction_days, int):
+            where_write_date_query = f"AND R.write_date >= CURRENT_DATE - interval '{self._rating_satisfaction_days} days'"
+        query = f"""
```

Comment:
Is the SQL necessary ? Did you run some tests indicating we cannot do sub-searches like
```
return [(
    'id', 'in',
    self.env['rating.rating'].sudo()._search([
        ('parent_res_id', '=', ...)
        ....
    ])
)]
```

Purpose would be to try to have readable code instead of manually hand crafted SQL.

Reply:
I understood but I don't know how to do the equivalent for this condition: `HAVING COALESCE(avg(R.rating), 0) {operator} %s`. 😢 
I could do something like this if you prefer: 
```py
import operator as py_operator
...


class RatingParentMixin(models.AbstractModel):
    ...

    def _search_rating_avg(self, operator, value):
        operator_mapping = {
            '=': py_operator.eq,
            '!=': py_operator.ne,
            '<': py_operator.lt,
            '<=': py_operator.le,
            '>': py_operator.gt,
            '>=': py_operator.ge,
        }
        if operator not in operator_mapping:
            raise NotImplementedError('This operator %s is not supported in this search method.' % operator)
        if not isinstance(value, (int, float)):
            raise NotImplementedError('The value must be an integer or a float value.')
        domain = [('parent_res_model', '=', self._name), ('consumed', '=', True), ('rating', '>=', RATING_LIMIT_MIN)]
        if self._rating_satisfaction_days:
            min_date = fields.datetime.now() - timedelta(days=self._rating_satisfaction_days)
            domain = expression.AND([domain, [('write_date', '>=', fields.Datetime.to_string(min_date))]])
        rating_read_group = self.env['rating.rating'].read_group(domain, ['parent_res_id', 'rating_avg:avg(rating)'], ['parent_res_id'])
        parent_res_ids = [res['parent_res_id'] for res in rating_read_group if operator_mapping[operator](res['rating_avg'], value)]
        return [('id', 'in', parent_res_ids)]
```

I tested and it works (perhaps you could improve this or do something else).

cc @IT-Ideas

Reply:
It reads nicer to me, as it is more standard domain / search odoo code... I generally avoid sql at all costs, unless there is a real performance issue that we cannot solve through ORM. Here I cannot really say, could be great to try it on our DB :D .

Reply:
I dont't think that it would make a big difference as it would just fetch some more records that we would have filtered previously with the HAVING in the SQL.
If you are happy with this and find it more readable it's all good for us!
And thanks for the review hein!